core/ledger/kvledger/kv_ledger.go

```diff
@@ -826,6 +830,52 @@ func (r *collectionInfoRetriever) CollectionInfo(chaincodeName, collectionName s
 	return r.infoProvider.CollectionInfo(r.ledgerID, chaincodeName, collectionName, qe)
 }
 
+type collectionConfigHistoryRetriever struct {
+	*confighistory.Retriever
+	ledger.DeployedChaincodeInfoProvider
+
+	ledger *kvLedger
+}
+
+func (r *collectionConfigHistoryRetriever) MostRecentCollectionConfigBelow(
+	blockNum uint64,
+	chaincodeName string,
+) (*ledger.CollectionConfigInfo, error) {
+	explicitCollections, err := r.Retriever.MostRecentCollectionConfigBelow(blockNum, chaincodeName)
+	if err != nil {
+		return nil, errors.WithMessage(err, "error while retrieving explicit collections")
+	}
+	qe, err := r.ledger.NewQueryExecutor()
+	if err != nil {
+		return nil, err
+	}
+	defer qe.Done()
+	implicitCollections, err := r.ImplicitCollections(r.ledger.ledgerID, chaincodeName, qe)
+	if err != nil {
+		return nil, errors.WithMessage(err, "error while retrieving implicit collections")
+	}
+
+	combinedCollections := explicitCollections
+	if combinedCollections == nil {
+		if implicitCollections == nil {
+			return nil, nil
+		}
```

Comment:
shouldn't we throw an error if an implicit collection is missing? In other words, it cannot be nil, right?

Reply:
This will be nil for a legacy chaincode - defined in the lscc.

Reply:
Ah. Thanks, Manish.